cmake/getvers.sh

```diff
@@ -1,12 +1,22 @@
 #!/bin/sh
 
-# Usage: getvers.sh <tree to get version info from> <output file>
+# Usage: getvers.sh <tree to get version info from> <build directory>
 
 set -e
+cd $1
 
-exec 1>$2
+# PEP440-compliant version number for pyproject.toml
+if [ -d .git ]; then
+	# replaces first - with .dev and second - with +, so e.g. 0.3-154-gd36baf4a becomes 0.3.dev154+gd36baf4a
+	fullversion_pep440=$(echo $(git describe --always --tags 2>/dev/null) | sed 's/-/.dev/' | sed 's/-/+/')
+fi
+fullversion_pep440="${fullversion_pep440:-0.1.0+unknown}" # fallback for SVN or error above
+sed "s/\\\$Version\\\$/$fullversion_pep440/" $1/cmake/pyproject.toml.in > $2/pyproject.toml
```

Comment:
Why do we want different representations of version numbers depending on how the software is installed?

Reply:
We don't necessarily want to. Right now its a git-based version, otherwise `-0.1.0+unknown`. Certainly if someone wanted to write something for CVS that'd be great, but maybe as a followup?

Reply:
Per discussion on Slack, the `fullversion` in `version.py` and version number in `pyproject.toml` is now the same, and its the `pyproject.toml`-style PEP440 compliant option, which constitutes a change to the former. 

(Small style note, I left this above the `exec 1>"$2/spt3g/version.py"` rather than in the if statement lower so that if it erroneously outputs anything that doesnt go into the file)